src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1808,6 +1956,105 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 	if (!is_target_path)
 		goto org_func;
 
+	/* Always rely on fuse for open() to avoid a fake fd */
+	if (compatible_mode) {
+		dfs_obj_t    *dfs_obj_local;
+		struct ht_fd *fd_ht_obj = NULL;
+		int           fd_fake;
+
+		if (two_args)
+			fd_kernel = real_open(pathname, oflags);
+		else
+			fd_kernel = real_open(pathname, oflags, mode);
+		/* dfuse fails for some reason */
+		if (fd_kernel < 0)
+			goto out_compatible;
+		/* query file object through ioctl() */
+		rc = fetch_dfs_cont_file_obj_with_fd(fd_kernel, dfs_mt, &dfs_obj_local);
+		if (rc)
+			goto out_compatible;
+		/* Need to create a fake fd and associate with fd_kernel */
+		atomic_fetch_add_relaxed(&num_open, 1);
+
+		rc = dfs_get_mode(dfs_obj_local, &mode_query);
+		if (rc)
+			goto out_compatible;
```

Comment:
do we need to close dfs_obj_local?
error ignored?

Reply:
You are right. "dfs_obj_local" should be closed. I will add a debug message when dfs_get_mode() fails. Thank you!